src/waveform/renderers/waveformmarkset.h

```diff
@@ -18,18 +18,18 @@ class WaveformMarkSet {
                const SkinContext& context,
                const WaveformSignalColors& signalColors);
 
-    int size() const { return m_marks.size();}
-    WaveformMarkPointer operator[] (int i) const { return m_marks[i]; };
+    inline QList<WaveformMarkPointer>::const_iterator begin() const { return m_marks.begin(); }
+    inline QList<WaveformMarkPointer>::const_iterator end() const { return m_marks.end(); }
 
     // hotCue must be valid (>= 0 and < NUM_HOT_CUES)
     WaveformMarkPointer getHotCueMark(int hotCue) const;
-    void setHotCueMark(int hotCue, WaveformMarkPointer pMark);
 
   private:
     void clear(){ m_marks.clear(); }
     std::unique_ptr<WaveformMark> m_pDefaultMark;
     QList<WaveformMarkPointer> m_marks;
-    int m_iFirstHotCue;
+    QMap<int, WaveformMarkPointer> m_hotCueMarks;
```

Comment:
Because it is filled randomly by the skin. Later the missing values are added. 
A QList solution may work as well if we change the filling loop and might perform better, but for now I like to keep the QMap with its integrated index safety.